js/isomer.js

```diff
@@ -67,25 +72,215 @@ Isomer.prototype._translatePoint = function (point) {
     }
   }
 };
 
 
+//+ Jonas Raoni Soares Silva
+//@ http://jsfromhell.com/math/is-point-in-poly [rev. #0]
+//see also http://jsperf.com/ispointinpath-boundary-test-speed/2
+function isPointInPoly(poly, pt){
+    for(var c = false, i = -1, l = poly.length, j = l - 1; ++i < l; j = i)
+        ((poly[i].y <= pt.y && pt.y < poly[j].y) || (poly[j].y <= pt.y && pt.y < poly[i].y))
+        && (pt.x < (poly[j].x - poly[i].x) * (pt.y - poly[i].y) / (poly[j].y - poly[i].y) + poly[i].x)
+        && (c = !c);
+    return c;
+}
+
+
+/**
+ * Does polygonA has intersection with polygonB ?
+ * Naive approach done first : approximate the polygons with a rectangle
+ * Then more complex method : see if edges cross, or one contained in the other
+ */
+Isomer.prototype._hasIntersection = function(pointsA, pointsB) {
+  var i, j;
+  var AminX = pointsA[0].x;
+  var AminY = pointsA[0].y;
+  var AmaxX = AminX;
+  var AmaxY = AminY;
+  var BminX = pointsB[0].x;
+  var BminY = pointsB[0].y;
+  var BmaxX = BminX;
+  var BmaxY = BminY;
+  for(i = 0 ; i < pointsA.length ; i++){
+    AminX = Math.min(AminX, pointsA[i].x);
+    AminY = Math.min(AminY, pointsA[i].y);
+	AmaxX = Math.max(AmaxX, pointsA[i].x);
+    AmaxY = Math.max(AmaxY, pointsA[i].y);
+  }
+  for(i = 0 ; i < pointsB.length ; i++){
+    BminX = Math.min(BminX, pointsB[i].x);
+    BminY = Math.min(BminY, pointsB[i].y);
+	BmaxX = Math.max(BmaxX, pointsB[i].x);
+    BmaxY = Math.max(BmaxY, pointsB[i].y);
+  }
+  
+  if(((AminX <= BminX && BminX <= AmaxX) || (BminX <= AminX && AminX <= BmaxX)) && 
+     ((AminY <= BminY && BminY <= AmaxY) || (BminY <= AminY && AminY <= BmaxY))) {
+    // now let's be more specific
+	var polyA = pointsA.slice();
+	var polyB = pointsB.slice();
+	polyA.push(pointsA[0]);
+	polyB.push(pointsB[0]);
+
+	// see if edges cross, or one contained in the other	
+	var deltaAX = [];
+	var deltaAY = [];
+	var deltaBX = [];
+	var deltaBY = [];
+	var rA = [];
+	var rB = [];
+	for(i = 0 ; i <= polyA.length - 2 ; i++){
+	  deltaAX[i] = polyA[i+1].x - polyA[i].x;
+	  deltaAY[i] = polyA[i+1].y - polyA[i].y;
+	  //equation written as deltaY.x - deltaX.y + r = 0
+	  rA[i] = deltaAX[i] * polyA[i].y - deltaAY[i] * polyA[i].x;
+	}
+	for(i = 0 ; i <= polyB.length - 2 ; i++){
+	  deltaBX[i] = polyB[i+1].x - polyB[i].x;
+	  deltaBY[i] = polyB[i+1].y - polyB[i].y;
+	  rB[i] = deltaBX[i] * polyB[i].y - deltaBY[i] * polyB[i].x;
+	}
+	
+	for(i = 0 ; i <= polyA.length - 2 ; i++){
+	  for(j = 0 ; j <= polyB.length - 2 ; j++){
+	    if(deltaAX[i] * deltaBY[j] != deltaAY[i] * deltaBX[j]){
+		  //case when vectors are colinear, or one polygon included in the other, is covered after
+		  //two segments cross each other if and only if the points of the first are on each side of the line defined by the second and vice-versa
+		  if((deltaAY[i] * polyB[j].x - deltaAX[i] * polyB[j].y + rA[i]) * (deltaAY[i] * polyB[j+1].x - deltaAX[i] * polyB[j+1].y + rA[i]) < -0.000000001 &&  
```

Comment:
These equations are inherently difficult to reason about, so at first glance it might be useful to either A) have a TON of comments or B) separating these lines into multiple statements, maybe some temporary variables?